File: qualifying/2106/Q1.c
#include <stdio.h>
#include<stdlib.h>

int smallest_pos_int(int arr[], int size){

    int pos_size = 0;
    for(int i=0; i<size; i++){
        if(arr[i] > 0){
            pos_size++;
        }
    }

    // printf("%d", pos_size);

    int* pos = (int *)malloc(pos_size*sizeof(int));
    int j=0;
    for(int i=0; i<size; i++){
        if(arr[i] > 0){
            pos[j] = arr[i];
            j++;
        }
    }

    int ans = pos_size+1;
    // printf("%d", ans);
    for(int i = 1; i <= pos_size; i++){
        int flag = 0;
        for (int j = 0; j < pos_size; j++){
            if(pos[j] == i){
                flag = 1;
            }
        }

        if (flag == 0){
            ans = i;
            break;
        }
    }
    free(pos);
    return ans;
}

int main(void){

    int arr1[] = {4, 7, -1, 1, 2};
    int arr2[] = {100, 101, 102};
    int arr3[] = {3, 2, 1, 0, -1};
    int arr4[] = {5, -1, 2, 4, 1, 3};

    printf("%d\n", smallest_pos_int(arr1, 5));
    printf("%d\n", smallest_pos_int(arr2, 3));
    printf("%d\n", smallest_pos_int(arr3, 5));
    printf("%d\n", smallest_pos_int(arr4, 6));

}

File: qualifying/2106/Q1.py
def smallest_pos_int(arr: list) -> int:
    ans = len(arr)
    for i in range(1, ans+1):
        if i not in arr:
            return i



if __name__ == "__main__":
    
    arr1 = [4, 7, -1, 1, 2]
    arr2 = [100, 101, 102]
    arr3 = [3, 2, 1, 0, -1]
    arr4 = [5, -1, 2, 4, 1, 3]

    print(f"Answer 1 : {smallest_pos_int(arr1)}")   # 3
    print(f"Answer 2 : {smallest_pos_int(arr2)}")   # 1
    print(f"Answer 3 : {smallest_pos_int(arr3)}")   # 4
    print(f"Answer 4 : {smallest_pos_int(arr4)}")   # 6